instrumentation/grape/lib/opentelemetry/instrumentation/grape/subscriber.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Grape
+      # Manages all subscriptions, both for custom subscribers and built-in notifications
+      class Subscriber
+        class << self
+          # Subscribe to all custom and built-in notifications (except those specified in the :ignored_events configs)
+          def subscribe
+            subscribe_to_custom_notifications
+            subscribe_to_built_in_notifications
+          end
+
+          private
+
+          SUBSCRIPTIONS = {
+            endpoint_render: 'endpoint_render.grape',
+            endpoint_run_filters: 'endpoint_run_filters.grape',
+            format_response: 'format_response.grape'
+          }.freeze
+
+          CUSTOM_SUBSCRIPTIONS = {
+            endpoint_run: 'OpenTelemetry::Instrumentation::Grape::CustomSubscribers::EndpointRun'
+          }.freeze
+
+          def subscribe_to_custom_notifications
+            CUSTOM_SUBSCRIPTIONS.each do |event, klass|
+              custom_subscriber = Object.const_get(klass).new
```

Comment:
Yes, I'd definitely prefer this ❤️ But there's a but 😅 
> I imagine this might enforce some requrie_relative or something in this class though.

Indeed I would need to add `require_relative 'custom_subscribers/endpoint_run'`. I'm fine with adding it inside a method to solve what @dhruvCW pointed out in [this comment](https://github.com/catawiki/opentelemetry-ruby-contrib/pull/3#discussion_r1142346023), but I'm afraid that wouldn't fix this issue either.

To be able to reference the class directly in the constant, the require should already be present when the class loads, and I don't think that will work here because of the way instrumentations are installed. I understand that all requires need to go through the install process to avoid loading files of instrumentations that are not installed.

Reply:
>I understand that all requires need to go through the install process to avoid loading files of instrumentations that are not installed.

I don't understand so if we define the require at the top of this file it would be called even if this file is not loaded ?

Reply:
I'm sure there's somestuff that could work with `autoload`, but I'm not sure of the usage in this repo (some people are very much against it because it's a mild perf overhead). IMO, it's fine the way it is, and I'd leave it up to the PR to the Otel people to see whether they have another pref.